### DISCUSSION TITLE
Can we update only npm for the selected node version with the use of nvm?

### DISCUSSION THREAD

**Discussion by parthtatva18** [coreybutler/nvm-windows#792] [Support]
Can we update only npm for the selected node version with the use of nvm?

**Comment by coreybutler** [comment]
NVM isn't required for this. Just run `npm install -g npm` to update npm. This only applies to the active version of Node. If you want to update npm for all versions of Node, you'd have to switch to each different version (i.e' `nvm use x.x.x`) and re-run `npm install -g npm` for each version you want to update.

**Reply by parthtatva18** [reply]
but when we are managing node environment with the use of nvm at that time `npm install -g npm`  wont work.

**Reply by coreybutler** [reply]
I don't think I completely understand what you're trying to do.

If you have Node 14.0.0 and want to upgrade npm, you'd first select the version by running `nvm use 14.0.0`. Next, you'd run `npm install -g npm`. This would update the version of npm only for the currently selected version of Node (14.0.0 in this example). Any other versions of Node which are already installed will be unaffected.

**Reply by parthtatva18** [reply]
![image](https://user-images.githubusercontent.com/46888427/164740256-3b5e1cc3-c403-4eaa-8750-4fcc814c05ce.png)

**Reply by coreybutler** [reply]
Ah - this is an upstream issue with specific versions of npm. In slightly older versions of npm, they hard-coded a path, preventing any tool from deleting npm files before installing new ones. This is, unfortunately, not something NVM4W (or other version managers) can control. In this particular case, a manual installation may be required.

Here you can see it works properly for a newer version of Node:

<img width="452" alt="image" src="https://user-images.githubusercontent.com/770982/164751221-615d2959-0a42-4033-888b-4a08454eb3d1.png">

Off the top of my head, I do not remember the specific versions of npm affected by this. With considerably older versions of node, it can fail for other reasons (as in newer npm doesn't support older versions of node).
